tools/signingtool/src/main/java/com/horizen/SigningToolCommandProcessor.java

```diff
@@ -0,0 +1,310 @@
+        ObjectNode resJson = new ObjectMapper().createObjectNode();
+
+        resJson.put("Error", error);
+        resJson.put("Usage", "validateMessage { \"message\": message, \"signature\": signature, \"publicKey\": public_key, \"prefix\": prefix, \"type\": \"string\" [\"schnorr\"] }");
+
+        printer.print(resJson.toString());
+    }
+
+    private void validateMessage(JsonNode json) {
+        if (!json.has("message") || !json.get("message").isTextual()) {
+            printValidateMessageUsageMsg("message is not specified or has invalid format.");
+            return;
+        } else if (!json.has("signature") || !json.get("signature").isTextual()) {
+            printValidateMessageUsageMsg("signature is not specified or has invalid format.");
+            return;
+        } else if (!json.has("publicKey") || !json.get("publicKey").isTextual()) {
+            printValidateMessageUsageMsg("publicKey is not specified or has invalid format.");
+            return;
+        } else if (!json.has("prefix") || !json.get("prefix").isTextual() || json.get("prefix").asText().isEmpty()) {
+            printValidateMessageUsageMsg("prefix is not specified or has invalid format.");
+            return;
+        } else if (!json.has("type") || !json.get("type").isTextual() || !Objects.equals(json.get("type").asText(), "schnorr")) {
+            printValidateMessageUsageMsg("type is not specified or has invalid format.");
+            return;
+        }
+        byte[] message = json.get("message").asText().getBytes();
+        String signature = json.get("signature").asText();
+        String publicKey = json.get("publicKey").asText();
+        byte[] prefix = json.get("prefix").asText().getBytes();
+        byte[] prefixMessage = hash(concatenate(prefix, message));
+
+        verifyMessage(signature, publicKey, prefixMessage);
+    }
+
+    private void printPrivKeyToPubKeyUsageMsg(String error) {
+        ObjectNode resJson = new ObjectMapper().createObjectNode();
+
+        resJson.put("Error", error);
+        resJson.put("Usage", "privKeyToPubKey { \"privateKey\": private_key, \"type\": \"string\" [\"schnorr\"] }");
+
+        printer.print(resJson.toString());
+    }
+
+    private void privKeyToPubKey(JsonNode json) {
+        if (!json.has("privateKey") || !json.get("privateKey").isTextual()) {
+            printPrivKeyToPubKeyUsageMsg("privateKey is not specified or has invalid format.");
+            return;
+        } else if (!json.has("type") || !json.get("type").isTextual() || !Objects.equals(json.get("type").asText(), "schnorr")) {
+            printPrivKeyToPubKeyUsageMsg("type is not specified or has invalid format.");
+            return;
+        }
+
+        String privateKey = json.get("privateKey").asText();
+        Secret secret = SECRETS_COMPANION.parseBytes(BytesUtils.fromHexString(privateKey));
+        assert secret.secretTypeId() == SecretsIdsEnum.SchnorrSecretKeyId.id();
```

Comment:
It seems that `assert` usage is too much here.
Much better to return a readable error: "Provided privateKey type is not supported...."
So the caller will be able to see the clear error message.
We should avoid processing errors in `processCommand` method.

Same in other places.